src/Algebra/Module/Construct/Nagata.agda

```diff
@@ -0,0 +1,170 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The non-commutative analogue of Nagata's construction
+-- of the "idealization of a module", defined here
+-- on R-R-*bi*modules M over a ring R.
```

Comment:
Link to paper?

Reply:
Do you mean: ourpaper, where I think this non-commutative generalisation first appears, or Nagata's first use in the late 1950s and his 1962 book "Local rings"?

Reply:
Whatever you think would be most helpful to the future code reader / maintainer to figure out your intent. And it's also nice to record the proper history (i.e. why is it called Nagata) here instead of forcing future readers to reconstruct.